web-client/public/js/upload.js

```diff
@@ -301,30 +329,104 @@ export const upload = function (sliderObject, sliderGroupController, drawGraph,
+    };
 
     $("#normalization-button").click(function () {
-        normalization = true;
-    // displayNetwork(currentNetwork, name, normalization);
-        drawGraph(currentNetwork, sliders, normalization, grayThreshold, nodeColoring);
+        synchronizeNormalizationValues($("#normalization-max").val());
+        drawGraph(currentNetwork, sliders, nodeColoring);
     });
 
-    $("#resetNormalizationButton").click(function () {
-        document.getElementById("normalization-max").value = "";
-    // normalization = false;
-    // displayNetwork(currentNetwork, name, normalization);
-        drawGraph(currentNetwork, sliders, normalization, grayThreshold, nodeColoring);
+    $("#reset-normalization-factor-menu, #resetNormalizationButton").click(function () {
+        synchronizeNormalizationValues("");
+        drawGraph(currentNetwork, sliders, nodeColoring);
     });
 
-    $("#grayThresholdInput").on("change", function () {
-        grayThreshold = true;
-    // displayNetwork(currentNetwork, name, normalization, grayThreshold);
-        drawGraph(currentNetwork, sliders, normalization, grayThreshold, nodeColoring);
+    $("#edge-weight-normalization-factor-menu").on("change", function () {
+        synchronizeNormalizationValues($("#edge-weight-normalization-factor-menu").val());
+        drawGraph(currentNetwork, sliders, nodeColoring);
     });
 
-    $("#dashedGrayLineButton").on("change", function () {
-    // displayNetwork(currentNetwork, name, normalization, grayThreshold);
-        drawGraph(currentNetwork, sliders, normalization, grayThreshold, nodeColoring);
+    var GREY_EDGE_THRESHOLD_MENU = "#gray-edge-threshold-menu";
+    var GREY_EDGE_THRESHOLD_SLIDER_SIDEBAR = "#grayThresholdInput";
+    var GREY_EDGE_THRESHOLD_TEXT_SIDEBAR = "#grayThresholdValue";
+    // var DEFAULT_GRAY_EDGE_THRESHOLD = 5;
+
+    // Gray Edge Controller
+
+    var grayEdgeInputValidator = function (value) {
+        if (value < 0) {
+            return 0;
+        } else if (value > 100) {
+            return 100;
+        } else {
+            return value;
+        }
+    };
+
+    var updateGrayEdgeValues = function (value) {
+        var validatedInput = grayEdgeInputValidator(value);
+        $(GREY_EDGE_THRESHOLD_TEXT_SIDEBAR).text(validatedInput + "%");
+        $(GREY_EDGE_THRESHOLD_MENU).val(validatedInput);
+        $(GREY_EDGE_THRESHOLD_SLIDER_SIDEBAR).val(validatedInput / 100);
+    };
+
+    $(GREY_EDGE_THRESHOLD_MENU).on("change", function () {
+        var value = Math.round(($(GREY_EDGE_THRESHOLD_MENU).val()));
+        updateGrayEdgeValues(value);
+        drawGraph(currentNetwork, sliders, nodeColoring);
+    });
+
+    $(GREY_EDGE_THRESHOLD_SLIDER_SIDEBAR).on("change", function () {
+        var value = Math.round(($(GREY_EDGE_THRESHOLD_SLIDER_SIDEBAR).val() * 100));
+        updateGrayEdgeValues(value);
+        drawGraph(currentNetwork, sliders, nodeColoring);
```

Comment:
`updateGrayEdgeValues` looks like another function that can just integrate `drawGraph` within itself, thus shortening the event handlers.

Reply:
Sounds good!